examples/react/react-app/src/App.tsx

```diff
@@ -116,7 +138,10 @@ function App() {
                     <button
                         onClick={() =>
                             position && position.vec.y > 0
-                                ? move(account.account, Direction.Up)
+                                ? dojo_starter.actions.move({
+                                      account: account.account.address,
+                                      args: [Direction.Up],
+                                  })
```

Comment:
```suggestion
                                ? dojo_starter.actions.move({
                                      account: account.account.address,
                                      args: [Direction.Up],
                                  })
```

i wonder if we could generate an interface that passes the args like a regular function. 

```ts
sdk.contract.fn(...args, config);

// or in this case
dojo_starter.action.move(Direction.Up, {
    account: account.account.address,
})
```

Reply:
I considered both cases during my first pass, but found that the object syntax for the call args felt like it made more sense.

Compare these two functions. In definition they don't seem to differ that much

```ts
const objectFunction = ({ direction, item, skill}: { direction: Direction, item: string, skill: string }) => {
    console.log(direction, item, skill);
}

const argsFunction = (direction: Direction, item: string, skill: string) => {
    console.log(direction, item, skill);
}
```

However in implementation, I find the object notation to provide way more clarity.

```ts
    objectFunction({
        direction: Direction.Up,
        item: "precious ring",
        skill: "blasting wand"
    });

    argsFunction(Direction.Up, "precious ring", "blasting wand");
```

Especially since every call will have a differing amount of args, this could be confusing.

Also say we would remove both the `precious ring` parameters, the error messages in the object notation are clearer imo.

![CleanShot 2024-03-11 at 10 12 49](https://github.com/dojoengine/dojo.js/assets/11869830/6446f448-c792-45cd-80f5-ef85a4697ac2)

vs 

![CleanShot 2024-03-11 at 10 13 03](https://github.com/dojoengine/dojo.js/assets/11869830/55dbfe35-c016-4976-aae6-5eece15167d2)

Reply:
Hmm I don't have a strong opinion but do prefer the second approach, since it is a direct translation from the contract abi. If a developer wanted the object args, they could pass a struct as input on the Cairo side and it would generate that for them right?

Reply:
Aha yeah, didn't think of that, makes sense!